### PR TITLE
[sfm] New sfmFilter software: split sfm based on filenames regex

### PR DESCRIPTION
**Pull request by jmelou**
SfMFilter can be used to create a new sfmdata containing a subset of views, selected using a regex on their name.
SfMTransfer can now select view using their poseID (used in MV-PS)

### CODE REVIEW

File: src/software/utils/main_sfmFilter.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        ("fileMatchingPattern,m", po::value<std::string>(&fileMatchingPattern)->required(),
+         "Matching pattern for the from_filepath method.\n")
+        ("outputSfMData_selected,o", po::value<std::string>(&outputSfMFilenameSelected)->required(),
+         "Path to the output SfMData file.\n")
+         ("outputSfMData_unselected,o", po::value<std::string>(&outputSfMFilenameUnselected)->required(),
+         "Path to the output SfMData file.\n");
+
+    CmdLine cmdline("AliceVision sfmFilter");
+    cmdline.add(requiredParams);
+    //cmdline.add(optionalParams);
```
Comment (cbentejac) [inline src/software/utils/main_sfmFilter.cpp]: Unused line, to be removed.

File: src/software/utils/main_sfmFilter.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    requiredParams.add_options()
+        ("inputFile,i", po::value<std::string>(&sfmDataFilename)->required(),
+         "Path to the input SfMData file.\n")
+        ("fileMatchingPattern,m", po::value<std::string>(&fileMatchingPattern)->required(),
+         "Matching pattern for the from_filepath method.\n")
+        ("outputSfMData_selected,o", po::value<std::string>(&outputSfMFilenameSelected)->required(),
+         "Path to the output SfMData file.\n")
+         ("outputSfMData_unselected,o", po::value<std::string>(&outputSfMFilenameUnselected)->required(),
+         "Path to the output SfMData file.\n");
+
```
Comment (cbentejac) [inline src/software/utils/main_sfmFilter.cpp]:
```suggestion
    // clang-format on
    
```

clang-format is disabled at the beginning of the parameters listing but it is never re-enabled. It should be re-enabled as soon as the parameters have been described.

File: src/software/utils/main_sfmFilter.cpp (hunk 1 above)
Comment (fabiencastan) [inline src/software/utils/main_sfmFilter.cpp]:
```suggestion
```

**Review by fabiencastan — APPROVED** [review]
Validating for now, but the SfMTransfer needs to be updated to propagate the landmarks with viewId associations.

**Pull request merged**